src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/reorder/reorder_kernel_base.cpp

```diff
@@ -175,13 +176,11 @@ ReorderKernelBase::DispatchData ReorderKernelBase::SetDefault(const reorder_para
                 break;
             }
         }
-    }
-
-    if ((params.output.GetLayout() == DataLayout::bs_fs_yx_bsv16_fsv16 ||
-         params.output.GetLayout() == DataLayout::bs_fs_yx_bsv32_fsv32 ||
-         params.output.GetLayout() == DataLayout::b_fs_yx_fsv16 ||
-         params.output.GetLayout() == DataLayout::bs_fs_yx_bsv32_fsv16) &&
-        params.inputs[0].Feature().v % 16 == 0) {
+    } else if ((output.GetLayout() == DataLayout::bs_fs_yx_bsv16_fsv16 ||
+                output.GetLayout() == DataLayout::bs_fs_yx_bsv32_fsv32 ||
+                output.GetLayout() == DataLayout::b_fs_yx_fsv16 ||
+                output.GetLayout() == DataLayout::bs_fs_yx_bsv32_fsv16) &&
+               input.Feature().v % 16 == 0 && dispatchData.gws[1] % 16 == 0) {
```

Comment:
just a question: is this to be applied only for rank size 4? e.g., b_fs_zyx_fsv16 layout is not expected to use this lws?

Reply:
I amn't sure about _b_fs_zyx_fsv16_ because my primary goal was to fix a bug inside existing logic. May be in some rare cases adding extra formats (e. g. _b_fs_zyx_fsv16_) to this list can improve performance. But when I made some perf experiments and tried different branches in the logic above I didn't see (in common) significant changes in per-layer timing statistics so I suppose that for most cases basic gws / lws combination calculated through _GetTensorFriendlyWorkGroups_ and _GetOptimalLocalWorkGroupSizes_ functions will work enough good.

So as a summary: may be, but it needs extra investigations.